psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -0,0 +1,766 @@
+#
+# @BEGIN LICENSE
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2007-2018 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This file is part of Psi4.
+#
+# Psi4 is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, version 3.
+#
+# Psi4 is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Psi4; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+"""
+The SCF iteration functions
+"""
+
+import numpy as np
+
+from psi4.driver import p4util
+from psi4.driver import constants
+from psi4.driver.p4util.exceptions import ConvergenceError, ValidationError
+from psi4 import core
+
+from .efp import get_qm_atoms_opts, modify_Fock_permanent, modify_Fock_induced
```

Comment:
`__all__` only controls `from psi4 import *`, I believe. `psi4.os` will still be accessible and in `dir(psi4)`. that's why I started deleting some temps https://github.com/psi4/psi4/blob/master/psi4/__init__.py#L73 . We can delete more. But since `__all__` doesn't actually restrict access, for us it only serves as guide-to-eye, and I think `public`/`_private` can do for that.

Another tidbit I found is that program chokes if you `from psi4 import *` and `__all__ = ['DO_NOT_WILD_IMPORT']` is present b/c that member is not in module. We could disable wild imports!

Reply:
Yea `__all__` will only effect `from psi4.submodule import *`, at the same time we do not say `from psi4.driver import os`. `os` comes from many places, I was mostly thinking that this particular would be a good spot to start the `__all__` revolution.